Update the header_with_proof test vectors after SSZ Union removal

kdeme opened the pull request.

acolytec3 reviewed (approved):
LGTM.  I ran this through our header validation logic and it's able to deserialize and verify the proofs for all of these pre-merge blocks.

njgheorghita commented:
LGTM! Confirmed that these works inside trin with our updated types

KolbyML reviewed (approved):
:shipit: when do you think we can merge this?

morph-dev commented:
I think that his PR should convert all affected test vectors from this repo. Otherwise, things would be very complicated...

The only other file that I found is (but I might have missed some):
[tests/mainnet/history/hive/test_data_collection_of_forks_blocks.yaml](https://github.com/ethereum/portal-spec-tests/blob/master/tests/mainnet/history/hive/test_data_collection_of_forks_blocks.yaml)

kdeme commented:
Updated the headers with proof  in the hive file in https://github.com/ethereum/portal-spec-tests/pull/34/commits/369b251f0e05fead302d1582684767982a6f6ce6, only pre-merge. Added a TODO for the rest, that should be done in https://github.com/ethereum/portal-spec-tests/pull/39 or in a new PR after that.
